src/audio/component.c

```diff
@@ -157,45 +157,6 @@ void comp_get_copy_limits(struct comp_buffer *source, struct comp_buffer *sink,
 	cl->sink_bytes = cl->frames * cl->sink_frame_bytes;
 }
 
-struct comp_dev *comp_make_shared(struct comp_dev *dev)
-{
```

Comment:
Hmm, a reference to comp_alloc() would be nice in the commit. I'm also a bit suspicious of the following comment in comp_alloc.c:

```
»       /*                                                                                                                                                                                                         
»        * use uncached address at the moment to rule out multi-core failures,                                                                                                                                     
»        * may need to switch to the latest coherence API for performance                                                                                                                                          
»        * improvement later.                                                                                                                                                                                      
»        */
»       dev = rzalloc(SOF_MEM_ZONE_RUNTIME_SHARED, 0, SOF_MEM_CAPS_RAM, bytes);
```

This smells a bit temporary in its wording, and by making comp_make_shared() a no-op in this PR, this in practise becomes permanent.

Reply:
as that comment says, if we ever decide to enable cached access to components, we'll need to use the coherence API. With that API you'd obtain a cached handle (pointer) to the object, use it, then release it again. So you would never re-link lists anyway, which is the large part of this function currently. I can modify that comment in `comp_alloc()`